#CNV-38162: new inferfromvolume content (#80363)
openshift/openshift-docs: modules/virt-using-flags-specify.adoc

// Module included in the following assemblies:
//
// * virt/virtual_machines/virt-specifying-instance-preference.adoc

:_mod-docs-content-type: PROCEDURE
[id="virt-using-flags-specify_{context}"]
= Using flags to specify instance types and preferences

> Specify both instance types and preferences by using flags.

Reviewer: ```suggestion
Specify instance types and preferences by using flags.
```

.Prerequisites

> * You must have an instance type, preference, or both, on the cluster.

Reviewer 2: ```suggestion
* You must have an instance type, preference, or both on the cluster.
```

.Procedure

. To specify an instance type when creating a VM, use the `--instancetype` flag. To specify a preference, use the `--preference` flag. The following example includes both flags:

+
[source,terminal]
----
$ virtctl create vm --instancetype <my-instancetype> --preference <my-preference>
----

> . Optional: To specify a namespaced instance type or preference, include the `kind` in the value passed to the `--instancetype` or `--preference` flag command. The namespaced instance type or preference must be in the same namespace you are creating the VM in. The following example includes both flags for a namespaced instance type and a namespaced preference:

Reviewer: ```suggestion
. Optional: To specify a namespaced instance type or preference, include the `kind` in the value passed to the `--instancetype` or `--preference` flag command. The namespaced instance type or preference must be in the same namespace you are creating the VM in. The following example includes flags for a namespaced instance type and a namespaced preference:
```

// Module included in the following assemblies:
//
// * virt/virtual_machines/virt-specifying-instance-preference.adoc

:_mod-docs-content-type: PROCEDURE
[id="virt-using-flags-specify_{context}"]
= Using flags to specify instance types and preferences

Specify both instance types and preferences by using flags.

.Prerequisites

* ???

.Procedure

To specify an instance type when creating a VM, use the `--instancetype` flag. To specify a preference, use the `--preference` flag. The following example includes both flags:

+
[source,terminal]
----
$ virtctl create vm --instancetype <my-instancetype> --preference <my-preference>
----

> Optional: To specify a namespaced instance type or preference, include the namespace when running the `create vm` command. The namespaced instance type or preference must be in the same namespace you are creating the VM in. The following example includes both flags for a namespaced instance type and namespaced preference:

Reviewer 2: ```suggestion
Optional: To specify a namespaced instance type or preference, include the kind in the valued passed to the `--instancetype` or `--preference` flag. The namespaced instance type or preference must be in the same namespace you are creating the VM in. The following example includes both flags for a namespaced instance type and namespaced preference:
```
